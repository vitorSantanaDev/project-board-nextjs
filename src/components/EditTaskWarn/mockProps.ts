export const handleClickCancelEdit = jest.fn()
