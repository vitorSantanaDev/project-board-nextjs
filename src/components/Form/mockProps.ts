export const formProps = {
  handleAddTaskSubmit: jest.fn(),
  handleChangeInput: jest.fn(),
  input: ''
}
