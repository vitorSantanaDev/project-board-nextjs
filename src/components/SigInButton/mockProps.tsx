export const mockProps = {
  session: true,
  onClick: jest.fn()
}
