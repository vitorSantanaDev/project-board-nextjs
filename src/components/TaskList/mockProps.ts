export const tasksProps = [
  {
    id: '3896433',
    userId: '8458',
    name: 'Name the User',
    tarefa: 'Name Task',
    created: new Date(),
    createdFormated: 'Date of Created'
  },
  {
    id: '39033',
    userId: '8456458',
    name: 'Name the User',
    tarefa: 'Name Task',
    created: new Date(),
    createdFormated: 'Date of Created'
  },
  {
    id: '38933',
    userId: '865658',
    name: 'Name the User',
    tarefa: 'Name Task',
    created: new Date(),
    createdFormated: 'Date of Created'
  },
  {
    id: '3366553',
    userId: '85654658',
    name: 'Name the User',
    tarefa: 'Name Task',
    created: new Date(),
    createdFormated: 'Date of Created'
  },
  {
    id: '337763',
    userId: '88',
    name: 'Name the User',
    tarefa: 'Name Task',
    created: new Date(),
    createdFormated: 'Date of Created'
  }
]
