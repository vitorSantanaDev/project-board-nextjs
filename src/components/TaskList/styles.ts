import styled from 'styled-components'

export const Form = styled.form`
  display: flex;
  align-items: center;
  justify-content: center;
`

export const InputElement = styled.input`
  width: 90%;
  height: 50px;

  padding: 8px 16px;

  border-radius: 6px;
  border: 1px solid #424242;

  color: #fff;
  background-color: #20212c;
`

export const ButtonSubmit = styled.button`
  width: 8%;
  height: 50px;
  cursor: pointer;
  border-radius: 6px;

  margin-left: 16px;

  background-color: #ffb800;
  transition: ease-in-out 0.3s;

  &:hover {
    transform: scale(1.05);
  }
`

export const Title = styled.h1`
  color: #fff;
  margin-top: 2.5rem;
`

export const SectionWrapper = styled.section``

export const TasklList = styled.article`
  margin: 2rem 0;
  padding: 1.2rem;
  border-radius: 6px;

  background-color: #20212c;

  > h3 {
    color: #f2f6fc;
    font-size: 1.6rem;
    line-height: 150%;

    cursor: pointer;
  }
`

export const ActionsWrapper = styled.div`
  width: 100%;

  margin-top: 2rem;

  display: flex;
  flex-direction: row;
  justify-content: space-between;

  div {
    display: flex;
    align-items: center;
    justify-content: center;

    time {
      color: #ffb800;

      margin-left: 0.5rem;
      margin-right: 1rem;
    }
  }
`

export const ButtonEdit = styled.button`
  border: none;
  background: transparent;

  cursor: pointer;
  margin-left: 1rem;
  transition: ease-in-out 0.3s;

  display: flex;
  align-items: center;
  justify-content: center;

  &:hover {
    opacity: 0.7;
    transform: scale(1.05);
  }

  > span {
    color: #fff;
    margin-left: 0.7rem;
  }
`

export const ButtonDelete = styled.button`
  border: none;
  background: transparent;

  cursor: pointer;
  margin-left: 1rem;
  transition: ease-in-out 0.3s;

  display: flex;
  align-items: center;
  justify-content: center;

  &:hover {
    opacity: 0.7;
    transform: scale(1.05);
  }

  > span {
    color: #fff;
    margin-left: 0.7rem;
  }
`
