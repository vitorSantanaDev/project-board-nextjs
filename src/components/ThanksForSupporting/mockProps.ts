export const mockProps = {
  lastDonateUser: `Sua última doação foi a ${new Date().toLocaleDateString()}`
}
