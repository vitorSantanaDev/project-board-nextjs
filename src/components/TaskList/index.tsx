import Link from 'next/link'
import { FiCalendar, FiEdit2, FiTrash } from 'react-icons/fi'
import * as S from './styles'

interface TaskListProps {
  tasks: {
    id: string
    userId: string
    name: string
    tarefa: string
    created: Date
    createdFormated?: string
  }[]
}

const TaskList: React.FC<TaskListProps> = ({ tasks }) => {
  return (
    <>
      {tasks.map((task) => (
        <S.SectionWrapper id="article" key={task.id}>
          <S.TasklList>
            <Link href={`/board/${task.id}`}>
              <h3>{task.tarefa}</h3>
            </Link>
            <S.ActionsWrapper>
              <div>
                <div>
                  <FiCalendar size={20} color="#FFB800" />
                  <time>{task.createdFormated}</time>
                </div>
                <S.ButtonEdit>
                  <FiEdit2 size={20} color="#FFF" />
                  <span>Editar</span>
                </S.ButtonEdit>
              </div>
              <S.ButtonDelete>
                <FiTrash size={20} color="#FF3636" />
                <span>Excluir</span>
              </S.ButtonDelete>
            </S.ActionsWrapper>
          </S.TasklList>
        </S.SectionWrapper>
      ))}
    </>
  )
}

export default TaskList
