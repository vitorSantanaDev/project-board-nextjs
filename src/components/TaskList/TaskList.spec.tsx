import { render, screen } from '@testing-library/react'
import { tasksProps } from './mockProps'
import TaskList from '.'

describe('<TaskList />', () => {
  it('should render the article', async () => {
    render(<TaskList tasks={tasksProps} />)
    const article = document.querySelector('#article')
    const headingTask = screen.getAllByRole('heading', { name: /name task/i })
    const createdFormated = screen.getAllByText('Date of Created')

    const editButtons = screen.getAllByRole('button', { name: /editar/i })
    const deleteButtons = screen.getAllByRole('button', { name: /excluir/i })

    expect(article).toBeInTheDocument()
    expect(headingTask).toHaveLength(5)
    expect(createdFormated).toHaveLength(5)
    expect(editButtons).toHaveLength(5)
    expect(deleteButtons).toHaveLength(5)

    screen.logTestingPlaygroundURL()
  })

  it('should match snapshot', () => {
    const { container } = render(<TaskList tasks={tasksProps} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
