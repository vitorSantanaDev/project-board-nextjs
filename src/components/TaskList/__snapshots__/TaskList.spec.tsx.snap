// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TaskList /> should match snapshot 1`] = `
.c0 {
  margin: 2rem 0;
  padding: 1.2rem;
  border-radius: 6px;
  background-color: #20212c;
}

.c0 > h3 {
  color: #f2f6fc;
  font-size: 1.6rem;
  line-height: 150%;
  cursor: pointer;
}

.c1 {
  width: 100%;
  margin-top: 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 div time {
  color: #ffb800;
  margin-left: 0.5rem;
  margin-right: 1rem;
}

.c2 {
  border: none;
  background: transparent;
  cursor: pointer;
  margin-left: 1rem;
  -webkit-transition: ease-in-out 0.3s;
  transition: ease-in-out 0.3s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2:hover {
  opacity: 0.7;
  -webkit-transform: scale(1.05);
  -ms-transform: scale(1.05);
  transform: scale(1.05);
}

.c2 > span {
  color: #fff;
  margin-left: 0.7rem;
}

.c3 {
  border: none;
  background: transparent;
  cursor: pointer;
  margin-left: 1rem;
  -webkit-transition: ease-in-out 0.3s;
  transition: ease-in-out 0.3s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3:hover {
  opacity: 0.7;
  -webkit-transform: scale(1.05);
  -ms-transform: scale(1.05);
  transform: scale(1.05);
}

.c3 > span {
  color: #fff;
  margin-left: 0.7rem;
}

<section
  class=""
  id="article"
>
  <article
    class="c0"
  >
    <h3>
      Name Task
    </h3>
    <div
      class="c1"
    >
      <div>
        <div>
          <svg
            color="#FFB800"
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            style="color: rgb(255, 184, 0);"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="18"
              rx="2"
              ry="2"
              width="18"
              x="3"
              y="4"
            />
            <line
              x1="16"
              x2="16"
              y1="2"
              y2="6"
            />
            <line
              x1="8"
              x2="8"
              y1="2"
              y2="6"
            />
            <line
              x1="3"
              x2="21"
              y1="10"
              y2="10"
            />
          </svg>
          <time>
            Date of Created
          </time>
        </div>
        <button
          class="c2"
        >
          <svg
            color="#FFF"
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            style="color: rgb(255, 255, 255);"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17 3a2.828 2.828 0 1 1 4 4L7.5 20.5 2 22l1.5-5.5L17 3z"
            />
          </svg>
          <span>
            Editar
          </span>
        </button>
      </div>
      <button
        class="c3"
      >
        <svg
          color="#FF3636"
          fill="none"
          height="20"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          style="color: rgb(255, 54, 54);"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="3 6 5 6 21 6"
          />
          <path
            d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
          />
        </svg>
        <span>
          Excluir
        </span>
      </button>
    </div>
  </article>
</section>
`;
